### PR TITLE
Add ability to run extra commands in the initialization jobs e.g. to quit istio sidecars

### PR DESCRIPTION
**Pull request by frankjkelly**
### Motivation

When using Istio sidecars they keep running even when a Kubernetes job has completed.
For example
```
$ kubectl get pods | grep pulsar | grep init
platform-pulsar-bookkeeper-init-vwp6v                             1/2     NotReady    0          13d
platform-pulsar-pulsar-init-c2c72                                 1/2     NotReady    0          13d
```

### Modifications

* Added a new Helm chart value `extraInitContainer` (default: "")
* This command can be used to send a quit call to Istio sidecars e.g.
```
  values:
    extraInitCommand: "curl -vvv -fsI -X POST http://localhost:15020/quitquitquit"
```
* See https://medium.com/redbox-techblog/handling-istio-sidecars-in-kubernetes-jobs-c392661c4af7 for more information including alternatives

### Local Testing 
Tested in Minikube where Istio is NOT running with
```
  values:
    extraInitCommand: "curl -vvv -fsI -X POST http://localhost:15020/quitquitquit"
```
and saw the following
```
15:55:22.416 [main-EventThread] INFO  org.apache.zookeeper.ClientCnxn - EventThread shut down for session: 0x100019308350014
15:55:22.416 [main] INFO  org.apache.zookeeper.ZooKeeper - Session: 0x100019308350014 closed
*   Trying 127.0.0.1:15020...
* connect to 127.0.0.1 port 15020 failed: Connection refused
*   Trying ::1:15020...
* Immediate connect fail for ::1: Cannot assign requested address
* Failed to connect to localhost port 15020: Connection refused
* Closing connection 0
```

also tested without the setting and the jobs started up fine with nothing new or unusual in the log files

```
$ kubectl get pods | grep pulsar
pulsar-bookkeeper-0                              1/1     Running     0          2m44s
pulsar-bookkeeper-init-b4r4w                     0/1     Completed   0          2m44s
pulsar-broker-0                                  1/1     Running     0          2m44s
pulsar-proxy-0                                   1/1     Running     0          2m44s
pulsar-pulsar-init-7zvc2                         0/1     Completed   0          2m44s
pulsar-zookeeper-0                               1/1     Running     0          2m44s
```

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

### CODE REVIEW

File: charts/pulsar/templates/bookkeeper-cluster-initialize.yaml [hunk 1]
```diff
@@ -69,6 +69,9 @@ spec:
                 {{- end }}
                 bin/bookkeeper shell initnewcluster;
             fi
+            {{- if .Values.istio.enabled }}
```
Comment (sijie) [inline charts/pulsar/templates/bookkeeper-cluster-initialize.yaml:72]: I would suggest a different approach:

Can you just introduce a setting called `extraInitCommand`? So you can include this in the init container. Then you can add this command to your values file. This provides better extensibility to the helm chart.

**Review by sijie — CHANGES_REQUESTED** [review]

**Review by sijie — APPROVED** [review]

**Review by michaeljmarshall — APPROVED** [review]

**Comment by frankjkelly** [comment]
Helm chart 2.7.7 was already released. See
https://github.com/apache/pulsar-helm-chart/releases/tag/pulsar-2.7.7
and
https://pulsar.apache.org/charts/index.yaml

**Review by michaeljmarshall — APPROVED** [review]
LGTM

**Review by lhotari — APPROVED** [review]
LGTM

**Pull request merged**